core/unix/coredump.c

```diff
@@ -479,26 +495,62 @@ os_dump_core_internal(dcontext_t *dcontext)
         return false;
     }
 
-    if (!write_elf_header(elf_file, /*entry_point=*/0,
+    if (!write_elf_header(elf_file, /*entry_point=*/get_entry_point(&mc),
                           /*section_header_table_offset*/ sizeof(ELF_HEADER_TYPE) +
-                              PROGRAM_HEADER_LENGTH + section_data_size,
+                              PROGRAM_HEADER_NOTE_LENGTH +
+                              sizeof(ELF_PROGRAM_HEADER_TYPE) * section_count +
+                              section_data_size,
                           /*flags=*/0,
-                          /*program_header_count=*/1,
+                          /*program_header_count=*/section_count,
```

Comment:
This is equal to the section count only because adjacent same-prot sections aren't merged yet?  Put a TODO here so that's clear.

Reply:
I've added a comment to explain how the number is computed.